src/main/java/juja/microservices/keepers/service/KeepersService.java

```diff
@@ -46,44 +44,51 @@
         if (keeper == null) {
-            logger.warn("Keeper with uuid '{}' and direction '{}' is't exist or inactive", uuid, direction);
+            logger.warn("Keeper with uuid '{}' and direction '{}' is't exist or not active", uuid, direction);
             throw new KeeperNonexistentException("Keeper with uuid " + uuid + " and direction " + direction
-                    + " is't exist or inactive");
+                    + " is't exist or not active");
         }
         keeper.setDismissDate(LocalDateTime.now());
-        List<String> ids = Collections.singletonList(keepersRepository.inactive(keeper));
-        logger.info("'Keeper' updated , with uuid {}, from user '{}'", uuid, form);
+        List<String> ids = Collections.singletonList(keepersRepository.save(keeper));
+        logger.info("'Keeper' deactivated , with uuid '{}', from user '{}'", uuid, from);
+        logger.debug("Keeper {} was deactivated ", uuid);
         return ids;
     }
 
     public String addKeeper(KeeperRequest keeperRequest) {
+        String uuid = keeperRequest.getUuid();
+        String direction = keeperRequest.getDirection();
+        String from = keeperRequest.getFrom();
         logger.debug("Service.addKeeper after in, parameters: {}", keeperRequest.toString());
-        if (keepersRepository.findOneByUUId(keeperRequest.getFrom()) == null) {
-            logger.warn("User '{}' tried to add new 'Keeper' but he is not a Keeper", keeperRequest.getFrom());
+        if (keepersRepository.findOneActive(from) == null) {
+            logger.warn("User '{}' tried to add new 'Keeper' but he is not a Keeper", from);
             throw new KeeperAccessException("Only the keeper can appoint another keeper");
         }
 
-        if (keepersRepository.findOneByUUIdAndDirectionIsActive(keeperRequest.getUuid(), keeperRequest.getDirection()) != null) {
-            logger.warn("Keeper with uuid '{}' already keeps direction '{}' and he is active",
-                    keeperRequest.getUuid(), keeperRequest.getDirection());
-            throw new KeeperDirectionActiveException("Keeper with uuid " + keeperRequest.getUuid() + " already keeps direction "
-                    + keeperRequest.getDirection() + " and he is active");
+        if (keepersRepository.findOneByUUIdAndDirectionIsActive(uuid, direction) != null) {
+            logger.warn("Keeper with uuid '{}' already keeps direction '{}' and he is active", uuid, direction);
+            throw new KeeperDirectionActiveException("Keeper with uuid " + uuid + " already keeps direction "
+                    + direction + " and he is active");
         }
-
-        String newKeeperId = keepersRepository.save(keeperRequest);
+        Date startDate = Date.from(LocalDateTime.now().atZone(ZoneId.systemDefault()).toInstant());
+        Keeper keeper = new Keeper(keeperRequest.getFrom(),
+                keeperRequest.getUuid(),
+                keeperRequest.getDirection(),
+                startDate);
```

Comment:
In rows 70-72 you already get all fields from KeeperRequest and asinged them to variables. Use these variables here.